### PR TITLE
Make CreateControlPlane use cloud-init

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
this PR completes the work started by @chuckha in https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/196 and makes CreateControlPlane use cloud-init + cleanups old code now replaced by CABPK.

With this change and some small fixes included in this PR, I managed to get the first control plane to start properly, but some additional work is still required in follow up PR to make the reconcile loop a little bit more stable and possibly aware of the external resources

**Release note**:
```release-note
NONE
```

/ assign @chuckha @detiber

### CODE REVIEW

File: docker/actions/kind.go [hunk 1]
```diff
@@ -274,40 +273,3 @@ func image(version string) string {
 	}
 	return defaults.Image
 }
-
-func removeExitedELBWithNameConflict(name string) error {
```
Comment (chuckha) [inline docker/actions/kind.go]: This wasn't used?

Comment (fabriziopandini) [reply docker/actions/kind.go]: It was used in create ELB, but I don't think we should keep this logic at this stage.
Eventually, we can add it back later...

File: docker/nodes.go [hunk 2]
```diff
@@ -143,17 +143,22 @@ func (n *Node) Create(cloudConfig []byte) (*nodes.Node, error) {
...
 			if err != nil {
-				log.Error(err, "Failed to get external load balancer node")
-				return nil, err
+				return nil, errors.Wrap(err, "failed to create control plane")
 			}
-			ipv4, _, err := node.IP()
+
+			// save the kubeconfig on the host with the loadbalancer endpoint
+			dest := actions.KubeConfigPath(n.Cluster)
+			hostAddress := "127.0.0.1"
```
Comment (chuckha) [inline docker/nodes.go]: Is this right? I think remoteutil will need the container address, not localhost

File: docker/actions/kind.go (hunk 1 above)
Comment (chuckha) [reply docker/actions/kind.go]: sgtm

File: docker/nodes.go (hunk 2 above)
Comment (fabriziopandini) [reply docker/nodes.go]: AFAIK as I know this writes the kubeconfig for the remote cluster on the host machine.

Writing the kubeconfig file in the management cluster is still an open point; see https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/143 and https://github.com/kubernetes-sigs/cluster-api/issues/1274

Comment (chuckha) [reply docker/nodes.go]: ah got it ok

**Comment by fabriziopandini** [comment]
@chuckha 
rebased and fixed the issue in verify-deps.sh. 
However, it is not fully clear to me why I should change deps in this PR...

File: docker/nodes.go (hunk 2 above)
Comment (chuckha) [reply docker/nodes.go]: Do you think we even need to do this? It seems like a nice-to-have versus something required for the system.

**Comment by chuckha** [comment]
You have to change deps because you deleted this https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/199/files#diff-2de70a80f649aa2d9a803e1d9108a712L26

This means we no longer (directly) import k/k, which is a good thing.

**Comment by fabriziopandini** [comment]
@chuckha thanks, great!
So this one is ready to go now...

**Pull request merged**